client/commands/v2/pysa_server.py

```diff
@@ -102,15 +169,22 @@ async def process_open_request(
         self, parameters: lsp.DidOpenTextDocumentParameters
     ) -> None:
         document_path = parameters.text_document.document_uri().to_file_path()
+        self.file_tracker[str(document_path)] = True
```

Comment:
Instead of tracking what files are open, why don't we track what files we've previously published diagnostics for? That would give us a much more exact picture of what files need their previous diagnostics overwritten. I imagine you could just update `file_tracker` in `update_errors` after this line:
```
diagnostics = self.invalid_models_to_diagnostics(model_errors)
```

Reply:
Would that be possible? Since the errors are being calculated for all files at once, how will we know which files to track here?